docs/proposals/13-automated-seed-management.md

```diff
@@ -0,0 +1,99 @@
+* By a special `gardenctl` command, for example `gardenctl drain <seed>`, that would automate the action described above. It should also add the `seed.gardener.cloud/unschedulable` taint to the specified seed.
+
+**Note:** The behavior described here is similar to the behavior of the `kubectl drain <node>` command. This command adds a taint with a `NoSchedule` effect to the node, but doesn't add any taints with a `NoExecute` effect. Instead, it simply forcibly evicts all pods from the node.
+
+### Scheduling Orphaned Shoots to Healthy Seeds 
+
+The next step in migrating shoots away from an unhealthy seed is rescheduling all orphaned shoots to healthy seeds by `gardener-scheduler`. The scheduler should ensure, among other things, that:
+
+* A shoot with the `orphan` annotation or with an empty `seedName` field is rescheduled.
+* A shoot is never scheduled on a seed tainted with a taint with a `NoSchedule` effect (unless the shoot tolerates the taint).
+* The new seed capacity for shoots is not exceeded by adding a new shoot, see [Ensuring Seeds Capacity for Shoots Is Not Exceeded](#ensuring-seeds-capacity-for-shoots-is-not-exceeded). This could perhaps also be achieved by automatically tainting the seed with a `NoSchedule` taint once its capacity is exceeded by the `gardenlet`.
+
+To reschedule a shoot, the scheduler updates the `seedName` field in the shoot specification with the name of the new seed. 
+
+### Migrating the Shoots to the New Seeds
+
+The final step in migrating shoots away from an unhealthy seed is actually migrating the shoots control planes from the old to the new seeds. The migration is triggered by the updates to the `seedName` field in the shoot specification by the scheduler described above.
+
+The migration process is described in more detail in [GEP 7: Shoot Control Plane Migration](07-shoot-control-plane-migration.md). 
+
+## Ensuring Seeds Capacity for Shoots Is Not Exceeded
+
+Another important automated seed management scenario is ensuring that a seed's capacity for shoots is not exceeded by introducing a maximum number of shoots that can be scheduled on a seed and making sure that it is taken into account by the scheduler. 
+
+An initial discussion of this topic is available in [Max cap for shoots per seed in scheduler](https://github.com/gardener/gardener/issues/2938). The proposed solution is based on the following flow:
+
+* The `gardenlet` is configured with certain *resources* and their *capacity*.
+* The `gardenlet` seed controller updates the seed status with the capacity of each resource and how much of it is currently available, using `capacity` and `allocatable` fields that are very similar to the corresponding fields in [the Node status](https://github.com/kubernetes/api/blob/2c3c141c931c0ab1ce1396c3152c72852b3d37ee/core/v1/types.go#L4582-L4593).
```

Comment:
The detection and publishing of whether the cluster is within or nearing its _capacity_ is relevant for even shoot clusters and I think can be solved in a way that would work for shoots as well as seeds. I like that you are using the analogy to node status. Could you also considering extending this to include conditions as in node status conditions? I had raised a [proposal long time ago](https://github.com/gardener/gardener/pull/1797) where there was an agreement on a subset of the proposal but never got around to follow-through.

Reply:
@amshuman-kr I don't see much overlap between your proposal (GEP-11) and the scope of the current proposal, which mainly concerns scheduling shoots onto seeds in a way that takes the seed capacity for various resources into account. Note that the scheduler, similarly to `kube-scheduler`, should only take into account the requested resource amounts, not the actually used amounts. I've updated the proposal now based on the last round of feedback, if you still think a reference to GEP-11 should be somehow included, could you please suggest a more concrete change?